assets/user_data.sh

```diff
@@ -40,6 +40,7 @@ function setup_hive_metastore {
       echo "UNATTACHED_VOLUME_ID: $UNATTACHED_VOLUME_ID"
 
       aws ec2 attach-volume --device "/dev/xvdh" --instance-id=$(ec2metadata --instance-id) --volume-id "$UNATTACHED_VOLUME_ID" --region ${aws_region}
+      # FIXME: This check is incompatible with bash's -e option. The attach-volume call either immediately succeeds or fails the whole script.
```

Comment:
I don't think we have had any problems with it, but if this is indeed a bug can you fix?

Reply:
Sorry, I didn't mean to commit this comment either. It was a note I made while debugging the EBS attachment failure. In the end it wasn't the cause of the issue. I'll revert it.

In my case an error occurred because the EBS volume had been created in a different availability zone from the EC2 instance and so could not be mounted. The incompatibility actually made it easier to catch the error because it failed once immediately instead of looping for ever.

If there are no problems in practice, I don't think we need to change it just now.

Before changing this I would have to understand why the EBS volume is created separately from the EC2 instance.